src/components/SearchResultItem.tsx

```diff
@@ -8,23 +8,94 @@ interface ParentTextClickedFunc {
 }
 
 interface SearchResultItemProps {
-    data: LibraryUtilities.ItemData;
-    libraryContainer: LibraryContainer;
-    highlightedText: string;
-    pathToItem: LibraryUtilities.ItemData[];
-    onParentTextClicked: ParentTextClickedFunc;
-    detailed: boolean;
+    data: LibraryUtilities.ItemData,
+    libraryContainer: LibraryContainer,
+    highlightedText: string,
+    pathToItem: LibraryUtilities.ItemData[],
+    onParentTextClicked: ParentTextClickedFunc,
+    detailed: boolean,
+    index: number
 }
 
-interface SearchResultItemStates { }
+interface SearchResultItemStates {
+    selected: boolean
+}
 
 export class SearchResultItem extends React.Component<SearchResultItemProps, SearchResultItemStates> {
 
     constructor(props: SearchResultItemProps) {
         super(props);
+        this.state = ({ selected: false });
+    }
+
+    componentWillMount() {
+        window.addEventListener("keydown", this.handleKeyDown.bind(this));
+    }
+
+    componentWillUnmount() {
+        window.removeEventListener("keydown", this.handleKeyDown.bind(this));
+    }
+
+    // Update selection state and scroll current item into view if the selected item is not in view yet.
+    componentDidUpdate() {
+        this.updateSelectionState();
+
+        if (this.state.selected) {
+            let container = ReactDOM.findDOMNode(this.props.libraryContainer);
+            let currentItem = ReactDOM.findDOMNode(this);
+            let containerRect = container.getBoundingClientRect();
+            let currentRect = currentItem.getBoundingClientRect();
+
+            if (currentRect.top < currentRect.height) {
+                currentItem.scrollIntoView();
+            }
+
+            if (currentRect.bottom > containerRect.bottom) {
+                currentItem.scrollIntoView(false);
+            }
+        }
+    }
+
+    handleKeyDown(event: any) {
```

Comment:
I think maybe the sequence that the event is called is the sequence that the event is registered. Since the items are generated by order, in my hands-on with it, the selection index is always updated as expected...